tests/unit_tests/test_game_rating.py

```diff
@@ -0,0 +1,695 @@
+import pytest
+from asynctest import CoroutineMock
+from collections import namedtuple
+import time
+import asyncio
+
+from server.rating import RatingType
+from trueskill import Rating
+from server.rating_service.rating_service import RatingService
+from server.rating_service.game_rater import GameRatingError
+
+from server.games.game import Game, GameState, GameError
+from server.games.game_results import GameOutcome
+from server.games.custom_game import CustomGame
+from server.games.ladder_game import LadderGame
+from tests.unit_tests.conftest import (
+    add_connected_player,
+    add_connected_players,
+    make_mock_game_connection,
+)
+
+pytestmark = pytest.mark.asyncio
+
+FFA_TEAM = 1
+
+
+class PersistenceError(Exception):
+    """
+    Raised when detecting that rating results would not have been persisted.
+    """
+
+    pass
+
+
+@pytest.fixture
+async def rating_service(database, player_service):
+    mock_service = RatingService(database, player_service)
+
+    mock_service._persist_rating_changes = CoroutineMock()
+
+    mock_ratings = {}
+
+    def set_mock_rating(player_id, rating_type, rating):
+        nonlocal mock_ratings
+        nonlocal mock_service
+        mock_service._logger.debug(
+            f"Set mock {rating_type} rating for player {player_id}: {rating}"
+        )
+        mock_ratings[(player_id, rating_type)] = rating
+
+    def get_mock_rating(*args, **kwargs):
+        nonlocal mock_ratings
+        nonlocal mock_service
+        player_id, rating_type = args
+        value = mock_ratings.get((player_id, rating_type), Rating(1500, 500))
+        mock_service._logger.debug(
+            f"Retrieved mock {rating_type} rating for player {player_id}: {value}"
+        )
+        return value
+
+    mock_service.set_mock_rating = set_mock_rating
+    mock_service._get_player_rating = CoroutineMock(wraps=get_mock_rating)
+
+    await mock_service.initialize()
+
+    yield mock_service
+
+    await mock_service.kill()
+
+
+def get_persisted_results(mock_service):
+    PersistedResults = namedtuple(
+        "PersistedResults", ["rating_type", "ratings", "outcomes"]
+    )
+    args = mock_service._persist_rating_changes.await_args
+    if args is None:
+        return PersistedResults(None, {}, {})
+
+    # TODO why does this need to be args[0]?
```

Comment:
Because mock `args` are a list of args for each time the mock was called. So `args[0]` gives you the args for the first call of the mock.